documents/entity/service.go

```diff
@@ -252,15 +265,31 @@ func (s service) GetVersion(ctx context.Context, documentID []byte, version []by
 	return s.get(ctx, documentID, version)
 }
 
+func (s service) GetEntityByRelationship(ctx context.Context, entityIdentifier, relationshipIdentifier []byte) (documents.Model, error) {
+	if s.Service.Exists(ctx, entityIdentifier) {
+		entity, err := s.Service.GetCurrentVersion(ctx, entityIdentifier)
+		if err != nil {
+			// in case of an error try to get document from collaborator
+			return s.RequestEntityWithRelationship(ctx, entityIdentifier, relationshipIdentifier)
+		}
+
+		// check if stored document is the latest version
+		if err := documents.PostAnchoredValidator(s.idService, s.anchorRepo).Validate(nil, entity); err != nil {
```

Comment:
Here you are implying that with any validation error (not explicitely the latest document) you will request the latest. Is this just a placeholder, until you add the latest logic in following PR?

Reply:
I opened a second PR for latestDocumentValidator.

We could only check the latest document or doing the complete `PostAnchorValidator` checking the roots etc.

Reply:
If we want to perform all validators is fine, but we need to only get the latest document when we failed checking the doc locally only on "not latest doc" validation error, not for all of them. At least at the moment.